test/CapabilitiesTest.php

```diff
@@ -98,6 +98,56 @@ public function testSilencedDeprecation(): void
         self::assertTrue(true);
     }
 
+    private function triggerWarning(): void
+    {
+        foreach (1 as $n) {}
+    }
+
+    public function testDuplicateWarningsA(): void
+    {
+        $this->triggerWarning();
+
+        self::assertTrue(true);
+    }
+
+    public function testDuplicateWarningsB(): void
+    {
+        $this->triggerWarning();
+        $this->triggerWarning();
+        $this->triggerWarning();
+
+        foreach (1 as $n) {}
+
+        self::assertTrue(true);
+    }
+
+    public function testDuplicateWarningsC(): void
+    {
+        self::assertTrue(true);
+    }
+
+    public function testMixedSeverities(): void
+    {
+        // Notice.
+        $foo = &self::provideData();
+        // Warning.
+        foreach (1 as $n) {}
+        // Deprecated.
+        trim(null);
+
+        self::assertTrue(true);
+    }
+
+    public function testSilencedWarningNotAffectsStatus(): void
+    {
+        $foo = @$bar;
```

Comment:
Is this a warning? If so, why not just reuse `triggerWarning()`?

Reply:
Just wondering why we haven't done the same for lines 134 and 119?

Reply:
Sorry, I missed your first comment's final version because GitHub doesn't seem to notify edits, so I only changed first line inside `testSilencedWarningNotAffectsStatus()`.

The `triggerWarning()` method was created because I needed to generate multiple warnings coming from the exact same location for "duplicate warnings.phpt". I'll use it to replace more inline warning generation, but the one in line 119 cannot be changed because it would defeat the purpose of asserting that it gets reported independently.

```php
    public function testDuplicateWarningsB(): void
    {
        $this->triggerWarning();
        $this->triggerWarning();
        $this->triggerWarning();

        foreach (1 as $n) {} # Line 119

        self::assertTrue(true);
    }
```

![image](https://github.com/user-attachments/assets/d5ee2488-f9fd-4743-80a0-5de00dbe2723)

Reply:
I cannot yet see changes here after 10 minutes, I suppose there's some delay, but inline warning in `testMixedSeverities()` is changed.

There're several other inline warnings, e.g. `testWarning()`. I don't know if you'd rather have them use the method. But there's no equivalent feature for other severities.